fedimint-bitcoind/src/electrum.rs

```diff
@@ -108,6 +108,45 @@ impl IBitcoindRpc for ElectrumClient {
         }
     }
 
+    async fn is_tx_in_block(
+        &self,
+        txid: &Txid,
+        block_hash: &BlockHash,
+        block_height: u64,
+    ) -> anyhow::Result<bool> {
+        let tx = block_in_place(|| self.0.transaction_get(txid))
+            .map_err(|error| info!(?error, "Unable to get raw transaction"));
+
+        match tx.ok() {
+            None => Ok(false),
+            Some(tx) => {
+                let output = tx
+                    .output
+                    // use last since that's the change output we've constructed
+                    .last()
+                    .ok_or(format_err!("Transaction must contain at least one output"))?;
+
+                let history = block_in_place(|| self.0.script_get_history(&output.script_pubkey))?;
+
+                // we check the earliest tx sent to our change address, since we can guarantee
+                // we constructed a withdrawal tx to our new change address but can't guarantee
+                // others have not sent subsequent transactions
```

Comment:
I don't think we can guarantee anything about the ordering. Anyone could race with it with a higher fee.

Reply:
Just `find` by `tx_hash` ? https://docs.rs/electrum-client/latest/electrum_client/struct.GetHistoryRes.html

Reply:
Ahh yes, that's a good point and I like your suggestion for using `find` to avoid the ordering issue.